viewing_party/party.py

```diff
@@ -1,23 +1,190 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if not title: 
+        return None
+    if not genre:
+        return None
+    if not rating:
+        return None
+    else: 
+        movie = {"title": title, "genre": genre, "rating" : rating}
+        return movie
+
+def add_to_watched(user_data, movie):
+    watched_movie = user_data["watched"]
+    watched_movie.append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist_movie = user_data["watchlist"]
+    watchlist_movie.append(movie)
+    
+    return user_data 
+
+
+
+#create function that takes two parameters
+def watch_movie(user_data, title):
+    user_watchlist = user_data["watchlist"]
+    user_watched = user_data["watched"]
+    for movie in user_watchlist:
+        if movie["title"] == title:
+            user_watchlist.remove(movie)
+            user_watched.append(movie)
+    return user_data
+    
+    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    watched = user_data["watched"]
+    avg_rating = []
+
+    while len(watched):
+        for movie in watched:
+            avg_rating.append(movie["rating"])
+            
+        average = sum(avg_rating) / len(avg_rating)
+        return average
+    return 0.0
+
+def get_most_watched_genre(user_data):
+    watched = user_data["watched"]
+    genre_list = []
+    freq_genre = {}
+
+    if len(watched):
+        
+        for movie in watched:
+            genre = movie["genre"]
+            freq_genre[genre] = 0
+            genre_list.append(genre)
+
+        for i in genre_list:
+            if i in freq_genre:
+                freq_genre[i] += 1
+        most_freq = max(freq_genre, key=freq_genre.get)
+        return most_freq
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):  
+    user_movie = []
+    friend_movie =[]
+    user_watched = user_data["watched"]
+    for movie in user_watched:
+        user_movie.append(movie)
+    
+
+    for friend in user_data["friends"]:
+        for title in friend["watched"]:
+            friend_movie.append(title)
+
+    for i in friend_movie:
+            if i in user_movie:
+                user_movie.remove(i)
+    return user_movie
+
+def get_friends_unique_watched(user_data):
+    user_movie = []
+    friend_movie = []
+    user_watched = user_data["watched"]
+    for movie in user_watched:
+        user_movie.append(movie)
+
+    friends_watched = user_data["friends"]
+    for movie in friends_watched:
+        for title in movie["watched"]:
+            if title in friend_movie:
+                continue
+            friend_movie.append(title)

```

Comment:
We can use the `not` operator to make this chunk of code one less line shorter (:

```python
    friends_watched = user_data["friends"]
    for movie in friends_watched:
        for title in movie["watched"]:
            if title not in friend_movie:
                friend_movie.append(title)
```